src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -106,10 +108,7 @@ internal sealed class TupleTypeSymbol : WrappedNamedTypeSymbol
             var constructedType = Create(underlyingType, elementNames, errorPositions, locationOpt, elementLocations);
             if (shouldCheckConstraints && diagnostics != null)
             {
-                // https://github.com/dotnet/roslyn/issues/33303
-                // Need to follow up on checking tuple constraints within a method body which may cause us to revisit the 
-                // explicit false below.
-                constructedType.CheckConstraints(compilation.Conversions, includeNullability: false, syntax, elementLocations, compilation, diagnostics);
+                constructedType.CheckConstraints(compilation.Conversions, includeNullability, syntax, elementLocations, compilation, diagnostics, diagnostics);
```

Comment:
>includeNullability [](start = 74, length = 18)

Is this no longer conditional on the language version?

Reply:
Determined by the caller. The one case in initial binding which still potentially passes `true` here though is when the nullable type is used explicitly in code. I assume, possibly wrong, that it should generate warnings in that case. 

---
In reply to: [257352693](https://github.com/dotnet/roslyn/pull/33336#discussion_r257352693) [](ancestors = 257352693)

Reply:
> Determined by the caller. The one case in initial binding which still potentially passes true here though is when the nullable type is used explicitly in code. I assume, possibly wrong, that it should generate warnings in that case. 

Yes, that feels inconsistent with other similar places where we report warnings outside of NullableWalker. Like for example in SourceMemberContainerSymbol_ImplementationChecks.cs, SourceComplexParameterSymbol.cs, TypeSymbol.cs, etc.

---
In reply to: [257406078](https://github.com/dotnet/roslyn/pull/33336#discussion_r257406078) [](ancestors = 257406078,257352693)

Reply:
sounds good. 

---
In reply to: [257409845](https://github.com/dotnet/roslyn/pull/33336#discussion_r257409845) [](ancestors = 257409845,257406078,257352693)